src/backend/core/tests/test_api_team_invitations.py

```diff
@@ -0,0 +1,299 @@
+    in teams they belong to, including from other issuers.
+    """
+    identity = factories.IdentityFactory()
+    other_user = factories.UserFactory()
+
+    team = factories.TeamFactory(
+        users=[(identity.user, "administrator"), (other_user, "owner")]
+    )
+    invitation = factories.InvitationFactory(
+        team=team, role="administrator", issuer=identity.user
+    )
+    invit2, invit3 = factories.InvitationFactory.create_batch(
+        2, team=team, role="member", issuer=other_user
+    )
+
+    # invitations from other teams should not be listed
+    other_team = factories.TeamFactory()
+    factories.InvitationFactory.create_batch(2, team=other_team, role="member")
+
+    client = APIClient()
+    client.force_login(identity.user)
+    response = client.get(
+        f"/api/v1.0/teams/{team.id}/invitations/",
+    )
+    assert response.status_code == status.HTTP_200_OK
+    assert response.json()["count"] == 3
+    assert sorted(response.json()["results"], key=lambda x: x["created_at"]) == sorted(
+        [
+            {
+                "created_at": invitation.created_at.isoformat().replace("+00:00", "Z"),
+                "email": str(invitation.email),
+                "team": str(team.id),
+                "role": invitation.role,
+                "issuer": str(invitation.issuer.id),
+                "is_expired": False,
+            },
+            {
+                "created_at": invit2.created_at.isoformat().replace("+00:00", "Z"),
+                "email": str(invit2.email),
+                "team": str(team.id),
+                "role": invit2.role,
+                "issuer": str(invit2.issuer.id),
+                "is_expired": False,
+            },
+            {
+                "created_at": invit3.created_at.isoformat().replace("+00:00", "Z"),
+                "email": str(invit3.email),
+                "team": str(team.id),
+                "role": invit3.role,
+                "issuer": str(invit3.issuer.id),
+                "is_expired": False,
+            },
+        ],
+        key=lambda x: x["created_at"],
+    )
+
+
+def test_api_team_invitations__list__all_invitations_are_listed(settings):
+    """
+    Both valid and expired invitations are listed.
+    """
+    identity = factories.IdentityFactory()
+    other_user = factories.UserFactory()
+
+    team = factories.TeamFactory(
+        users=[(identity.user, "administrator"), (other_user, "owner")]
+    )
+
+    # override settings to accelerate validation expiration
+    settings.INVITATION_VALIDITY_DURATION = 3  # second
+    expired_invit = factories.InvitationFactory(
+        team=team,
+        role="member",
+        issuer=identity.user,
+    )
+    time.sleep(3)
+
+    valid_invit = factories.InvitationFactory(
+        team=team, role="administrator", issuer=identity.user
+    )
+
+    client = APIClient()
+    client.force_login(identity.user)
+    response = client.get(
+        f"/api/v1.0/teams/{team.id}/invitations/",
+    )
```

Comment:
- I can't set up `created_at` easily, 
- changing INVITATION_VALIDITY_DURATION impact both invitations
- I didn't want to use an additional package (such as [freezegun](https://github.com/spulec/freezegun/blob/master/README.rst) for instance) to avoid an extra dependancy solely for this test 

so I chose a super naive approach. I suppose, in theory, on very old machines, test could fail because `valid_invitation` becomes invalid in the time we query the endpoint but I don't think I could really happen neither on local nor on our servers.

Reply:
I would recommend you to add [freezegun](https://github.com/spulec/freezegun/blob/master/README.rst)

Reply:
Dates are a complex thing, and expiration data a critical part. We should test them properly, feel free to use the appropriate tool for it.

Reply:
or, you could try to mock the default function that give created_at its value. I am not sure it would work.

Reply:
I realized what we really wanted to test here is that expired invitations don't disappear and are still listed correctly. I don't need to tweak spacetime to do that. I slightly modified the test and it's robust now, I don't think I need to add freezegun or anything else.